### ISSUE TITLE
Docker creates new volume instead of use one from volumes-from after upgrading from 1.2.0 to 1.3.2

### ISSUE THREAD

**Issue by grossws** [docker/docker#9768]
I use data-only container pattern with data-only volume with bind-mount. E. g., I create data-only container with 
`docker run --name crowd-postgres-data -v /opt/crowd-postgres-data:/var/lib/postgresql/data postgres:9.3.5 /bin/true` 
and start postgres with 
`docker run -d --name crowd-postgres.service --volumes-from crowd-postgres-data -p 5432:5432 -e LANG=C.UTF-8 -e POSTGRES_PASSWORD=xxx postgres:9.3.5`.

It works fine on docker 1.2.0/1.14. Same pattern works on docker 1.3.2/1.15 only on *new* data-only container. It breaks on old data-only container with new app container. 
If I try to start postgres with second command it receives usual volume (from `/var/lib/docker/vfs/dir`) instead of volumes from container selected by `--volumes-from` arg, so all data seems lost.

Results of `docker inspect` on these containers available in [gist](https://gist.github.com/grossws/6526f6d0bb7df861c614). 

Interesting parts: 
- https://gist.github.com/grossws/6526f6d0bb7df861c614#file-postgres-data-L100
- https://gist.github.com/grossws/6526f6d0bb7df861c614#file-postgres-L77
- https://gist.github.com/grossws/6526f6d0bb7df861c614#file-postgres-L115

This behavior persists after rebooting etc. Also, nothing special in docker logs, no dm errors.

Workaround: remove data-only container and create it again. If data-only container is created with same docker version as app-container bug doesn't manifest itself.

Env:
```
# cat /etc/centos-release 
CentOS Linux release 7.0.1406 (Core)

# uname -a
Linux localhost 3.10.0-123.9.3.el7.x86_64 #1 SMP Thu Dec 18 20:10:00 MSK 2014 x86_64 x86_64 x86_64 GNU/Linux

# docker version 
Client version: 1.3.2
Client API version: 1.15
Go version (client): go1.3.3
Git commit (client): 39fa2fa/1.3.2
OS/Arch (client): linux/amd64
Server version: 1.3.2
Server API version: 1.15
Go version (server): go1.3.3
Git commit (server): 39fa2fa/1.3.2

# docker -D info
Containers: 26
Images: 278
Storage Driver: devicemapper
 Pool Name: docker-253:3-134320238-pool
 Pool Blocksize: 65.54 kB
 Data file: /var/lib/docker/devicemapper/devicemapper/data
 Metadata file: /var/lib/docker/devicemapper/devicemapper/metadata
 Data Space Used: 6.278 GB
 Data Space Total: 107.4 GB
 Metadata Space Used: 11.07 MB
 Metadata Space Total: 2.147 GB
 Library Version: 1.02.84-RHEL7 (2014-03-26)
Execution Driver: native-0.2
Kernel Version: 3.10.0-123.9.3.el7.x86_64
Operating System: CentOS Linux 7 (Core)
Debug mode (server): false
Debug mode (client): true
Fds: 17
Goroutines: 18
EventsListeners: 0
Init SHA1: c906504aa058139c1d0569ecd0aa5f462a73440f
Init Path: /usr/libexec/docker/dockerinit
```

**Comment by cpuguy83** [comment]
Seems like the volumes aren't being registered with the daemon for some reason.

Maybe when iterating through the registered containers on daemon restart we're missing some?

**Comment by cpuguy83** [comment]
Resolved by #10147
